File: app/src/main/java/com/lkms/data/model/java/Mention.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "Mention".
 * Bảng này có khóa chính phức hợp (commentId, userId).
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Mention {

    // Khóa ngoại đến Comment
    @SerializedName("commentId")
    private Integer commentId;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;
}

File: app/src/main/java/com/lkms/data/model/Mention.kt
// Trong file: data/model/Mention.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "Mention".
 * Bảng này có khóa chính phức hợp (commentId, userId).
 */
@Serializable
data class Mention(
    // Khóa ngoại đến Comment
    @SerialName("commentId")
    val commentId: Int,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int
)